115-update-KafkaMirrorMaker2-resource-to-better-correspond-its-use.md

```diff
@@ -0,0 +1,212 @@
+However, this whole section will be optional in `v1beta2`, so this will not break backwards compatibility, and users who do not use the `.spec.target` section do not care about any required fields in it.
+To allow early migration to the new API while the `v1beta2` API is still in use, the `.spec.connectCluster` field will no longer be required in `v1beta2` (it is completely removed in `v1`).
+
+The following example shows the new `KafkaMirrorMaker2` layout:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker-2
+spec:
+  version: 4.0.0
+  replicas: 1
+  target:
+    alias: east
+    bootstrapServers: east-kafka-bootstrap:9092
+    groupId: my-mm2-group
+    configStorageTopic: my-mm2-config
+    statusStorageTopic: my-mm2-status
+    offsetStorageTopic: my-mm2-offsets
+    config:
+      config.storage.replication.factor: -1
+      offset.storage.replication.factor: -1
+      status.storage.replication.factor: -1
+  mirrors:
+    - source:
+        alias: "west"
+        bootstrapServers: west-kafka-bootstrap:9092
+      sourceConnector:
+        tasksMax: 1
+        config:
+          replication.factor: -1
+          offset-syncs.topic.replication.factor: -1
+          sync.topic.acls.enabled: "false"
+          refresh.topics.interval.seconds: 600
+      checkpointConnector:
+        tasksMax: 1
+        config:
+          checkpoints.topic.replication.factor: -1
+          sync.group.offsets.enabled: "false"
+          refresh.groups.interval.seconds: 600
+      topicsPattern: ".*"
+      groupsPattern: ".*"
+    - source:
+        alias: "south-pole"
+        bootstrapServers: south-pole-kafka-bootstrap:9092
+      sourceConnector:
+        tasksMax: 1
+        config:
+          replication.factor: -1
+          offset-syncs.topic.replication.factor: -1
+          sync.topic.acls.enabled: "false"
+          refresh.topics.interval.seconds: 600
+      checkpointConnector:
+        tasksMax: 1
+        config:
+          checkpoints.topic.replication.factor: -1
+          sync.group.offsets.enabled: "false"
+          refresh.groups.interval.seconds: 600
+      topicsPattern: ".*"
+      groupsPattern: ".*"
+```
+
+The new fields will be available to Strimzi users starting with Strimzi 0.49.0.
+When both the new and old fields are configured, Strimzi will always prefer the configuration from the new fields.
+During this phase, we would expect the custom resource to use either the new or the old API.
+The old configuration will remain fully supported as long as we support the `v1beta2` API.
+Once we drop support for the `v1beta2` API, we will also clean the legacy code from the Strimzi Cluster Operator and support only the new API.
+
+Users who decide not to migrate and use the new fields early will migrate to them when moving to the `v1` API.
```

Comment:
This sentence slightly confuses me, are we saying, if users do not migrate but use the new fields as well as the old fields?